include/fiction/utils/layout_utils.hpp

```diff
@@ -224,6 +230,123 @@ template <typename Lyt>
+        {
+            if (c.y <= y_offset)
+            {
+                y_offset = c.y;
+            }
+            if (c.x <= x_offset)
+            {
+                x_offset = c.x;
+            }
+        });
+
+    lyt.foreach_cell(
+        [&lyt_new, &lyt, &x_offset, &y_offset](const auto& c)
+        {
+            lyt_new.assign_cell_type({c.x - x_offset, c.y - y_offset}, lyt.get_cell_type(c));
+            lyt_new.assign_cell_mode({c.x - x_offset, c.y - y_offset}, lyt.get_cell_mode(c));
+            lyt_new.assign_cell_name({c.x - x_offset, c.y - y_offset}, lyt.get_cell_name(c));
+        });
+
+    return lyt_new;
+}
+
+/**
+ * Converts the coordinates of a given cell-level layout to SiQAD coordinates. A new equivalent layout based on SiQAD
+ * coordinates is returned.
+ *
+ * @tparam Lyt Cell-level layout type based on fiction coordinates, e.g., `offset::ucoord_t` or `cube::coord_t`.
+ * @param lyt The layout that is to be converted to a new layout based on SiQAD coordinates.
+ * @return A new equivalent layout based on SiQAD coordinates.
+ */
+template <typename Lyt>
+sidb_cell_clk_lyt_siqad convert_to_siqad_coordinates(const Lyt& lyt) noexcept
+{
+    static_assert(is_cartesian_layout_v<Lyt>, "Lyt is not a Cartesian layout");
+    static_assert(is_cell_level_layout_v<Lyt>, "Lyt is not a cell-level layout");
+    static_assert(has_sidb_technology_v<Lyt>, "Lyt is not an SiDB layout");
+
+    sidb_cell_clk_lyt_siqad lyt_new{{lyt.x(), lyt.y()},
+                                    lyt.get_layout_name(),
+                                    lyt.get_tile_size_x(),
+                                    lyt.get_tile_size_y()};
+
+    lyt.foreach_cell(
+        [&lyt_new, &lyt](const auto& c)
+        {
+            lyt_new.assign_cell_type(siqad::to_siqad_coord<cell<Lyt>>(c), lyt.get_cell_type(c));
+            lyt_new.assign_cell_mode(siqad::to_siqad_coord<cell<Lyt>>(c), lyt.get_cell_mode(c));
+            lyt_new.assign_cell_name(siqad::to_siqad_coord<cell<Lyt>>(c), lyt.get_cell_name(c));
+        });
+
+    return lyt_new;
+}
+
+/**
+ * Converts the coordinates of a given cell-level layout to fiction coordinates, e.g., `offset::ucoord_t` or
+ * `cube::coord_t`. A new equivalent layout based on fiction coordinates is returned.
+ *
+ * @tparam Lyt Cell-level layout type based on fiction coordinates.
+ * @param lyt The layout that is to be converted to a new layout based on fiction coordinates.
+ * @return A new equivalent layout based on fiction coordinates.
+ */
+template <typename Lyt>
+Lyt convert_to_fiction_coordinates(const sidb_cell_clk_lyt_siqad& lyt) noexcept
+{
+    static_assert(is_cartesian_layout_v<Lyt>, "Lyt is not a Cartesian layout");
+    static_assert(is_cell_level_layout_v<Lyt>, "Lyt is not a cell-level layout");
+    static_assert(has_sidb_technology_v<Lyt>, "Lyt is not an SiDB layout");
+
+    Lyt lyt_new{{lyt.x(), lyt.y()}, lyt.get_layout_name(), lyt.get_tile_size_x(), lyt.get_tile_size_y()};
```

Comment:
I am not quite sure how best to implement the aspect ratio change due to normalization. Would it make sense to add a method in ``cell-level-layout'' to set the aspect ratio?

Reply:
Every coordinate layout is required to have a `resize()` function that sets a new aspect ratio

Reply:
Thanks!